tests/test_settings.py

```diff
@@ -69,6 +71,47 @@ def test_flat_config(self, flat_settings: SettingsExample) -> None:
         assert flat_settings.settings_folder == Path.home()
         assert Path(Path.home() / self.SETTINGS_FILE_NAME).exists()
 
+    def test_xdg_config(self, xdg_settings: SettingsExample) -> None:
+        """Test that settings file is created in the xdg_config_home folder."""
+        assert xdg_settings.settings_folder.exists()
+        assert xdg_settings.settings_folder.is_dir()
+        assert xdg_settings.settings_folder.name == f"{self.TEST_APP_NAME}"
+        assert xdg_settings.settings_file_name == self.SETTINGS_FILE_NAME
+        assert (
+            xdg_settings.settings_folder
+            == xdg_config_home() / f"{self.TEST_APP_NAME}"
+        )
+
+        assert xdg_settings.get("app_name") == "test_app"
+        assert xdg_settings.get("test_string_var") == "test_value"
+
+    def test_settings_from_environment(self) -> None:
+        """Test that the settings file is loaded from the xdg variable."""
+        xdg_orig_value = os.environ.get("XDG_CONFIG_HOME", None)
+        home_path = Path.home()
+        expected_path = home_path / ".config"
+        default_path = xdg_config_home()
+        assert default_path == expected_path
+
+        expected_path = home_path / "/path/validity/matters/not"
+        assert os.environ.get("XDG_CONFIG_HOME_ALT") == str(expected_path)
+
+        # alternate path does not exist
+        os.environ.setdefault(
+            "XDG_CONFIG_HOME", os.environ["XDG_CONFIG_HOME_ALT"]
+        )
+        modified_xdg_path = xdg_config_home()
+        assert modified_xdg_path != expected_path
+        assert modified_xdg_path == default_path
+
+        # alternate path does exist
+        os.environ.setdefault("XDG_CONFIG_HOME", str(home_path))
+        assert modified_xdg_path == xdg_config_home()
+
+        # reset to pre-test env value
+        if xdg_orig_value:
+            os.environ.setdefault("XDG_CONFIG_HOME", xdg_orig_value)
+
```

Comment:
For this test, we dont need the `pytest-env` package as that changes the env var for ALL tests, which may later be an issue. 

Just use the built-in `monkeypatch` functionality. I'm not using the xdg_settings` fixture, since that is created BEFORE we can patch the env var, so we just quickly make a new settings instance instead. You can also remove the extra package from `pyproject.toml` (and the env settings)

The original test failed for me as it was written too, maybe different setups?🤷‍♂️

```suggestion
    def test_settings_from_environment(self, monkeypatch, fs) -> None:
        """Test that the settings file is read from the xdg variable."""
        monkeypatch.setenv("XDG_CONFIG_HOME", "/path/validity/matters/not")

        settings = SettingsExample("test_app", xdg_config=True)
        assert (
            settings.settings_folder
            == Path("/path/validity/matters/not") / "test_app"
        )
```